### PR TITLE
[dev-docs] add releasing.md

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by jmarshall** [comment]
It might be worth adding that if hand-creating a tag, it should ideally be an annotated (or signed) tag, similar to the CI-generated ones.

**Comment by danking** [comment]
I think CI's *tags* are actually not signed, but its commits are. We never hand create main commits, so those should continue to be signed. The main commits are signed because GitHub does the merges for us. We should set up a GPG key for CI and sign our tags too. That would be best.

**Review by daniel-goldstein — DISMISSED** [review]

**Comment by danking** [comment]
> CI's tags are annotated, i.e., they have their own datestamp and message. The recent 0.2.128 tag is a lightweight tag, so for example `git describe` of current HEAD still says 0.2.127-87-ge002b4b23; you have to add `--tags` to get 0.2.128-8-ge002b4b23. Not a biggie, but worth adding to a checklist.
> 
> But yes, eventually, signing them would probably be best practice.

Ah, I see, thanks for pushing on this. I was not aware there were two kinds of tags. I'll fix the 0.2.128 tag.

File: dev-docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    cat contents/password | skopeo login --username hailgenetics --password-stdin docker.io
+
+Next we need to cook up a valid pypirc:
+
+    download-secret hail-ci-0-1-github-oauth-token
+    cp contents/pypirc $HOME/.pypirc
+
+Next we need to cook up a valid github-oauth token (for creating GitHub releases):
+
+    printf 'Authorization: token ' > /PATH/TO/WORKING_DIR/$BUILD_TOKEN/github-oauth
```
[the remark this answers is not in the document]
Comment (danking) [reply dev-docs/releasing.md:71]: I decided to save the path to the directory of artifacts & etc. in an envvar.

**Comment by danking** [comment]
@jmarshall thanks again for the feedback 0.2.128 should now be fixed: https://github.com/hail-is/hail/releases/tag/0.2.128 .

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**